d3ploy/supply_driven_deployment_inst.py

```diff
@@ -229,6 +259,11 @@ def decision(self):
                     str(self.commodity_supply[commod][time]) + "\n"
                 with open(commod + ".txt", 'a') as f:
                     f.write(out_text)
+        for child in self.children:
```

Comment:
I wonder if we use a local variable keep track of the exit times for each children so this loop isn't done each time step. It probably doesn't take too much computation time, but might be a cleaner implementation?

Reply:
I think a loop at every time step is inevitable because we need to keep track of every single facility that newly enters the simulation.

Reply:
I was thinking a local variable something like a map of {time:[agent, agent]}

Then just having it check if the current time step is in the keys and decommission the agents in the array. It's probably not that much faster and realistic is more memory intensive. 

I withdraw my comment.